src/script.ts

```diff
@@ -6,7 +6,7 @@
 
   const knownColorSchemes = ['dark', 'light']
 
-  const preference = window.localStorage.getItem('<%= options.storageKey %>') || '<%= options.preference %>'
+  const preference = window?.localStorage?.getItem('<%= options.storageKey %>') || '<%= options.preference %>'
```

Comment:
```suggestion
  const preference = (window.localStorage && window.localStorage.getItem('<%= options.storageKey %>')) || '<%= options.preference %>'
```

It might be better to do such, as optionally chaining may not be supported in all browsers

Reply:
checkout the last commit, I think it is better like that

Reply:
@antfu what you think, does it make sense to encapsulate this into a try catch ? laoding the module in incognito iframe in cross domain does actually brake the whole page
the initial load should be 100% safe imo

Reply:
any hurdles for this to get merged ? I would be happy to contribute